src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationSource.cs

```diff
@@ -31,32 +28,39 @@ public IConfigurationProvider Build(IConfigurationBuilder builder)
             try
             {
                 AzureAppConfigurationOptions options = _optionsProvider();
-                ConfigurationClient client;
+                IConfigurationClientManager clientManager;
 
-                if (options.Client != null)
+                if (options.ClientManager != null)
                 {
-                    client = options.Client;
+                    clientManager = options.ClientManager;
                 }
                 else if (!string.IsNullOrWhiteSpace(options.ConnectionString))
                 {
-                    client = _configurationClientFactory.CreateConfigurationClient(options.ConnectionString, options.ClientOptions);
+                    clientManager = new ConfigurationClientManager(options.ConnectionString, options.ClientOptions);
                 }
-                else if (options.Endpoint != null && options.Credential != null)
+                else if (options.Endpoints != null && options.Credential != null)
                 {
-                    client = _configurationClientFactory.CreateConfigurationClient(options.Endpoint, options.Credential, options.ClientOptions);
+                    clientManager = new ConfigurationClientManager(options.Endpoints, options.Credential, options.ClientOptions);
                 }
                 else
                 {
                     throw new ArgumentException($"Please call {nameof(AzureAppConfigurationOptions)}.{nameof(AzureAppConfigurationOptions.Connect)} to specify how to connect to Azure App Configuration.");
                 }
 
-                provider = new AzureAppConfigurationProvider(client, options, _optional);
+                provider = new AzureAppConfigurationProvider(clientManager, options, _optional);
             }
-            catch (InvalidOperationException e)
+            catch (InvalidOperationException ex) // InvalidOperationException is thrown when any problems are found while configuring AzureAppConfigurationOptions.
```

Comment:
SDK also throws this exception when it fails to create a new ConfigurationClient. For example, if the connection string in invalid, SDK will throw `InvalidOperationException`  on `new ConfigurationClient(...)` call.

Reply:
Thanks for pointing out!